File: backend/src/main/java/ma/marocsphere/service/AvisService.java
package ma.marocsphere.service;

import ma.marocsphere.dto.AvisCreationDTO;
import ma.marocsphere.dto.AvisResponseDTO;

import java.util.List;

public interface AvisService {
    AvisResponseDTO getById(Long id);
    List<AvisResponseDTO> getByGuideId(Long guideId);
    List<AvisResponseDTO> getByArtisanId(Long artisanId);
    AvisResponseDTO create(AvisCreationDTO dto);
}

File: frontend/src/app/services/avis.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { API_BASE_URL } from '../config/api.config';
import { Avis } from '../models/api.models';

@Injectable({ providedIn: 'root' })
export class AvisService {
  private http = inject(HttpClient);
  private readonly baseUrl = `${API_BASE_URL}/avis`;

  getByGuideId(guideId: number) {
    return this.http.get<Avis[]>(`${this.baseUrl}/guide/${guideId}`);
  }

  getByArtisanId(artisanId: number) {
    return this.http.get<Avis[]>(`${this.baseUrl}/artisan/${artisanId}`);
  }

  getById(id: number) {
    return this.http.get<Avis>(`${this.baseUrl}/${id}`);
  }

  create(data: { auteurId: number; cibleId?: number; artisanId?: number; note: number; commentaire: string }) {
    return this.http.post<Avis>(this.baseUrl, data);
  }
}
